src/coordination/coordinator_instance.cpp

```diff
@@ -557,5 +560,65 @@ auto CoordinatorInstance::IsReplica(std::string_view instance_name) const -> boo
   return raft_state_.IsReplica(instance_name);
 }
 
+auto CoordinatorInstance::GetRoutingTable(std::map<std::string, std::string> const &routing) -> RoutingTable {
+  auto res = RoutingTable{};
+
+  // TODO: (andi) Test if there is not current main, what routig table returns... Unit test because no user involvement
+  // Also test drivers
+  // TODO: (andi) Disable this when used on replicaton instance
+  auto const is_instance_alive = [&](ReplicationInstanceState const &instance) {
+    auto const &repl_instance = FindReplicationInstance(instance.config.instance_name);
+    return repl_instance.IsAlive();
+  };
+
+  auto const repl_instance_to_bolt = [](ReplicationInstanceState const &instance) {
+    return instance.config.BoltSocketAddress();
+  };
+
+  auto const is_instance_main = [&](ReplicationInstanceState const &instance) {
+    return instance.status == ReplicationRole::MAIN;
+  };
+
+  auto const is_instance_replica = [&](ReplicationInstanceState const &instance) {
+    return instance.status == ReplicationRole::REPLICA;
+  };
+
+  auto const &raft_log_repl_instances = raft_state_.GetReplicationInstances();
+
+  auto bolt_mains = raft_log_repl_instances | ranges::views::filter(is_instance_alive) |
+                    ranges::views::filter(is_instance_main) | ranges::views::transform(repl_instance_to_bolt) |
+                    ranges::to<std::vector>();
+  MG_ASSERT(bolt_mains.size() <= 1, "There can be at most one main instance active!");
```

Comment:
I would just add here todo, because is_instance_main is still wrong check. We will have multiple MAINs still, but with `uuid` we will know which one is correct one.
I am handling that in #1819 PR